tests/unit/test_lib/test_utilities/test_filters/test_filters.py

```diff
@@ -0,0 +1,207 @@
+#  - Redistributions of source code must retain the above copyright notice,   #
+#      this list of conditions and the following disclaimer.                  #
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import time
+import unittest
+
+from liota.lib.utilities.filters.range_filter import RangeFilter, Type
+from liota.lib.utilities.filters.windowing_scheme.windowing_scheme import WindowingScheme
+
+
+class FilterTest(unittest.TestCase):
+    """
+    Range Filter Unit Test Cases
+    """
+    def setUp(self):
+        self.lower_bound = 10
+        self.upper_bound = 20
+        self.window_test_filter = RangeFilter(Type.CLOSED, self.lower_bound, self.upper_bound)
+
+    """
+    Range filter initialization tests
+    """
+    def test_init_filter_type (self):
+        self.assertRaises(ValueError, lambda : RangeFilter(None, 0, 1))
+
+    def test_init_lower_upper_bound(self):
+        self.assertRaises(TypeError, lambda: RangeFilter(Type.CLOSED, "0", "1"))
+
+    def test_init_lower_bound(self):
+        self.assertRaises(TypeError, lambda: RangeFilter(Type.CLOSED, "0", 1))
+
+    def test_init_upper_bound(self):
+        self.assertRaises(TypeError, lambda : RangeFilter(Type.CLOSED, 1, "2"))
+
+    def test_init_lower_less_than(self):
+        self.assertRaises(ValueError, lambda: RangeFilter(Type.LESS_THAN, "0", 1))
+
+    def test_init_upper_greater_than(self):
+        self.assertRaises(ValueError, lambda: RangeFilter(Type.GREATER_THAN, 1, "2"))
+
+    """
+    Accept filters : Bounded at both end
+    Values : Ranging inside limit
+    """
+    def test_number_closed(self):
```

Comment:
Add comment that, "if filter is applied over values that are not of type `Number`, the values will be returned without applying filter.  Nice testcase :)